powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/CreateActivationRequest.java

```diff
@@ -31,20 +35,53 @@
 @Data
 public class CreateActivationRequest {
 
+    @Schema(description = "The identifier of the user")
+    @NotBlank(message = "User ID must not be empty when creating activation")
     private String userId;
+
+    @Schema(description = "Timestamp of activation expiration")
+    @Future(message = "The activation expiration timestamp must be in the future when creating activation")
     private Date timestampActivationExpire;
+
+    @Schema(description = "Whether recovery codes should be generated")
     private boolean generateRecoveryCodes = true;
+
+    @Schema(description = "Maximum number of failures for the activation")
+    @Positive(message = "Maximum failure count must be positive when creating activation")
     private Long maxFailureCount;
+
+    @Schema(description = "Application key")
+    @NotBlank(message = "Application key must not be empty when creating activation")
     private String applicationKey;
+
+    @Schema(description = "Identifier of the temporary key for encryption")
     private String temporaryKeyId;
+
+    @Schema(description = "Ephemeral public key used in encryption")
+    @NotBlank(message = "Ephemeral public key must not be empty when creating activation")
     private String ephemeralPublicKey;
+
+    @Schema(description = "Encrypted data used in encryption")
```

Comment:
The second part of the sentence doesn't sound right.
```suggestion
    @Schema(description = "Encrypted data")
```